streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/utils/TestDriver.scala

```diff
@@ -29,7 +29,7 @@ import org.scalatest.Suite
 
 trait TestDriver { this: Suite =>
 
-  def createTestDriver(builder: StreamsBuilder, initialWallClockTimeMs: Long = System.currentTimeMillis()) = {
+  def createTestDriver(builder: StreamsBuilder, initialWallClockTimeMs: Long = 0): TopologyTestDriver = {
```

Comment:
sorry, two changes on this line:
* default to `0` instead of current time: this is the fix
* add return value `TopologyTestDriver`: this resolves a warning that public methods should have explicit return types.

Reply:
Can you elaborate on the `initialWallClockTimeMs` fix? Why is using current time a problem?

Reply:
We are seeing the KStream test fail to find join results sometimes. The fact that it doesn't happen all the time suggests that there's some non-determinism in the test. Using the current system time is definitely a source of non-determinism, so I figured we could start by eliminating the obvious.

Reply:
I am not sure what test you refer to, however, the result should only depend on the record timestamps -- not the driver wall-clock time. Or does the test use wall-clock-time punctuation? If this is the case, should the test not specify the time the test want to use and initial value instead of changing the test driver?

Just try to understand the scope.

Reply:
This is fair. After discussion, I don't think this particular line is the problem. I'll revert it.